### PR TITLE
[5.4] Contextual binding works on the bound aliase issue #19225

### PR DESCRIPTION
**Pull request by LeoYang90**
here is my test code:
```php
public function testContextualBindingWorksOnBoundAliase()
{
    $container = new Container;

    $container->alias('Illuminate\Tests\Container\IContainerContractStub','stub');
    $container->bind('stub', ContainerImplementationStub::class);

    $container->when('Illuminate\Tests\Container\ContainerTestContextInjectOne')->needs('stub')->give('Illuminate\Tests\Container\ContainerImplementationStubTwo');

    $this->assertInstanceOf(
        'Illuminate\Tests\Container\ContainerImplementationStubTwo',
        $container->make('Illuminate\Tests\Container\ContainerTestContextInjectOne')->impl
    ); 
}
```

According to my understanding, if ‘ stub’ is bound to another class, it can not alias to or represent the class 'Illuminate\Tests\Container\IContainerContractStub' anymore. However, in the case, 'stub' can still work in contextual binding in needs().

```php
class ContainerTestContextInjectOne
{
    public $impl;

    public function __construct(IContainerContractStub $impl)
    {
        $this->impl = $impl;
    }
}

interface IContainerContractStub
{
}

class ContainerImplementationStub implements IContainerContractStub
{
}

class ContainerImplementationStubTwo implements IContainerContractStub
{
}
```
More detail information is in issue #19225.
thanks!

### CODE REVIEW

**Comment by taylorotwell** [comment]
Your test is hard to follow and has spelling errors. I don't know why you are using a try / catch. Please re-write to be followable.

**Comment by LeoYang90** [comment]
@taylorotwell 
Ok, I get it. Thank you!
